### PR TITLE
Issue #158 - Add "Claim your prize" link 🐺`

### PR DESCRIPTION
**Pull request by FrancescoSTL**
Issue ref: #158 

@schalkneethling r?

### CODE REVIEW

File: data/css/all-aboard.css [hunk 1]
```diff
@@ -57,7 +57,7 @@ a:focus {
 
 .sidebar-content main,
 .sidebar-content footer {
-    padding: 20px 40px 40px 40px;
+    padding: 20px 40px 20px 40px;
```
Comment (schalkneethling) [inline data/css/all-aboard.css]: You can shorten this to just `padding: 20px 40px;`

File: data/js/utility/content5.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    //unhide "claim your prize!"
+    if (document.querySelector('.prize')) {
+        var prizeLink = document.querySelector('.prize');
+        prizeLink.classList.remove('hidden');
+
+        prizeLink.addEventListener('click', function() {
+        	addon.port.emit('intent', 'claimPrize');
+        });
+    }
+})();
```
Comment (schalkneethling) [inline data/js/utility/content5.js]: So, for the utility track, the user gets the badge on step5 just for opening the sidebar so, we can just have the prize link visible by default. We would therefore only need lines 4, and 7-9 here.

File: data/tmpl/utility/content5.html [hunk 3]
```diff
@@ -49,7 +49,11 @@
           </li>
         </ul>
       </footer>
+      <div class="prize hidden">
+        <a class="prize">Claim your prize!</a>
```
Comment (schalkneethling) [inline data/tmpl/utility/content5.html]: This should be a `button` element, as it does something and not send you somewhere.

File: data/tmpl/utility/content5.html [hunk 4]
```diff
@@ -49,7 +49,11 @@
           </li>
         </ul>
       </footer>
+      <div class="prize hidden">
```
Comment (schalkneethling) [inline data/tmpl/utility/content5.html]: We do not need to hide this. Also, do we *need* the wrapping `div`?

File: data/tmpl/values/content5.html [hunk 5]
```diff
@@ -72,6 +72,9 @@
           </li>
         </ul>
       </footer>
+      <div class="prize hidden">
+        <a class="prize">Claim your prize!</a>
```
Comment (schalkneethling) [inline data/tmpl/values/content5.html]: Same here, use `button` element.

File: index.js [hunk 6]
```diff
@@ -365,6 +365,13 @@ function intentHandler(intent) {
         case 'highlightURL':
             highLight('urlbar');
             break;
+        case 'claimPrize':
+            var sidebarProps = {};
+            sidebarProps.id = "all-aboard-reward";
+            sidebarProps.title = "Prize";
+            sidebarProps.url = "./tmpl/reward.html";
+            showSidebar(sidebarProps);
```
Comment (schalkneethling) [inline index.js]: Hmmmm, `showSidebar` has all kinds a logic that are aimed at the general 5 content sidebars. Things such as workers listening for `cta_complete`, `intent`(which is fine here), and `show`. It also does things like set the step, update the `distribution.id` and also emits the `tokens` event.

I therefore do not think it is a good idea to re-use it here. It would be preferable to create a `showRewardSidebar` function that is specifically made for this. We can then also use that function when showing this after the reward notification.

**Comment by FrancescoSTL** [comment]
Fixed the things 🏄 

@schalkneethling r?

**Comment by FrancescoSTL** [comment]
Alas, I've jumped the gun

File: data/js/utility/content5.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+(function() {
+    var prizeLink = document.querySelector('.prize');
+    
+    prizeLink.addEventListener('click', function() {
+       	addon.port.emit('intent', 'claimPrize');
```
Comment (schalkneethling) [inline data/js/utility/content5.js]: Nit: indentation

File: data/tmpl/utility/content5.html [hunk 8]
```diff
@@ -49,7 +49,11 @@
           </li>
         </ul>
       </footer>
+      <button class="prize">
```
Comment (schalkneethling) [inline data/tmpl/utility/content5.html]: Could we just use `class="button"` here?

**Comment by chrismore** [comment]
I am reviewing the functionality of this now

**Comment by chrismore** [comment]
Appears to be working as planned.

**Comment by chrismore** [comment]
Tested this both on the values and utility path and both of the links appear after you either answer the question or see the mobile sidebar.

There is currently no 1/12th notification of the reward sidebar as described in the #159

**Comment by FrancescoSTL** [comment]
@chrismore yep, still working on the timer/notification

**Comment by FrancescoSTL** [comment]
@chrismore done with notification timer. r?

**Comment by chrismore** [comment]
Testing now

**Comment by chrismore** [comment]
@FrancescoSTL got the notification in 1/12th the time, but i clicked the notification with the existing sidebar open and it closed the sidebar instead of refreshing the content.

**Comment by FrancescoSTL** [comment]
@chrismore can you replicate and let me know the track/ steps? I'll take a look shortly, working out the values layout

**Comment by FrancescoSTL** [comment]
@chrismore ready for review again!

**Comment by chrismore** [comment]
Tested it both on the values and utility track and both timers work and the sidebar stays open if already open.

**Comment by FrancescoSTL** [comment]
Gotta merge this for the funnelcake. @schalkneethling you can retroactively review if you'd like :)
